src/index.ts

```diff
@@ -0,0 +1,83 @@
+// tslint:disable:no-console
+import {
+  Beautifier,
+  Language,
+  BeautifierBeautifyData,
+  DependencyType,
+  NodeDependency,
+} from "unibeautify";
+import * as cosmiconfig from "cosmiconfig";
+import * as readPkgUp from "read-pkg-up";
+import options from "./options";
+
+const { pkg } = readPkgUp.sync({ cwd: __dirname });
+export const beautifier: Beautifier = {
+  name: "CSScomb",
+  package: pkg,
+  dependencies: [
+    {
+      type: DependencyType.Node,
+      name: "CSScomb",
+      package: "csscomb",
+    },
+  ],
+  options: {
+    CSS: options.CSS,
+    SCSS: options.CSS,
+    Sass: options.CSS,
+  },
+  resolveConfig: ({ filePath, projectPath, dependencies }) => {
+    const pathToSearch = filePath ? filePath : projectPath;
+    if (!pathToSearch) {
+      return Promise.resolve({});
+    }
+    const Comb = dependencies.get<NodeDependency>("CSScomb").package;
+    const explorer = cosmiconfig("csscomb", {
+      searchPlaces: [
+        ".csscomb.json",
+        "csscomb.json",
+        ".csscomb.js",
+        "csscomb.js",
+      ],
+    });
+    return explorer
+      .search(pathToSearch)
+      .then(result => {
+        if (!result) {
+          return Promise.resolve({
+            config: Comb.detectInFile(pathToSearch),
+          });
+        } else {
+          return Promise.resolve({
+            config: result.config,
+          });
+        }
+      })
+      .catch(error => {
+        // tslint:disable-next-line:no-console
+        console.log(error);
+        return Promise.resolve({});
+      });
+  },
+  beautify({
+    text,
+    options,
+    filePath,
+    projectPath,
+    dependencies,
+    beautifierConfig,
+  }: BeautifierBeautifyData) {
+    return new Promise<string>((resolve, reject) => {
+      const CSScomb = dependencies.get<NodeDependency>("CSScomb").package;
+      const config = (beautifierConfig && beautifierConfig.config) || {};
+      const comb = new CSScomb(config);
+      const result = comb.processString(text);
+      if (result) {
+        return resolve(result);
+      } else {
+        return resolve(text);
```

Comment:
@szeck87 We may want beautifiers to actually throw in these cases, so we know if something went wrong. I recall a lot of issues in Atom-Beautify where users are wondering whether a beautifier is working and we wouldn't want the beautifier to report a success when it was really a failure. Something to consider for later.

Reply:
What would we throw though?  It's not an error since the beautifier works, there just isn't anything for it to do.

Reply:
We should return an object with the returning text and some additional info; did the text successfully beautified: true/false and an error message if there was one. So we could process them in plugins like VS Code to display some information as a notification or in the logs.

Reply:
@muuvmuuv  Either

```javascript
throw new Error("the error message");
```

or

```
return Promise.reject(new Error("the error message"));
```

should be sufficient.

Reply:
@szeck87 I think:

- there will always be a result output even if there is nothing to do for the beautifier when a file is already fine
- no result would mean there has been no text processed that could be beautified for the engine -> so it has failed somewhere

I'll try to set up some tests for that